service/matching/taskListManager.go

```diff
@@ -423,6 +422,10 @@ getTasksPumpLoop:
 					logging.LogPersistantStoreErrorEvent(c.logger, logging.TagValueStoreOperationUpdateTaskList, err,
```

Comment:
Actually it is a good opportunity to cleanup logging for all tasklist life-cycle.  You may want to model it similar to shard life-cycle to make debugging easier.